BrowserKit/Sources/SummarizeKit/SummarizeController.swift

```diff
@@ -0,0 +1,314 @@
+        transformAnimation.fromValue = 0
+        transformAnimation.toValue = frameHeight / 2
+        transformAnimation.duration = UX.initialTransformAnimationDuration
+        transformAnimation.timingFunction = UX.initialTransformTimingCurve
+        transformAnimation.fillMode = .forwards
+        tabSnapshot.layer.add(transformAnimation, forKey: "translation")
+        tabSnapshot.transform = CGAffineTransform(translationX: 0.0,
+                                                  y: view.frame.height * UX.tabSnapshotInitialTransformPercentage)
+
+        gradient.animatePositionChange(animationCurve: UX.initialTransformTimingCurve)
+
+        UIView.animate(withDuration: UX.initialTransformAnimationDuration, delay: 0.0, options: [], animations: {
+            self.tabSnapshot.layer.cornerRadius = self.screenCornerRadius
+            self.loadingLabel.alpha = 1.0
+        }) { _ in
+            // This is here just to show case the full animation
+            DispatchQueue.main.asyncAfter(deadline: .now() + 4.0) {
+                self.showSummary()
+            }
+        }
+    }
+
+    private func showSummary() {
+        let theme = themeManager.getCurrentTheme(for: currentWindowUUID)
+        tabSnapshot.isUserInteractionEnabled = true
+        tabSnapshot.addGestureRecognizer(UIPanGestureRecognizer(target: self, action: #selector(onTabSnapshotPan)))
+        let tabSnapshotOffset = tabSnapshotTopConstraint?.constant ?? 0.0
+
+        UIView.animate(withDuration: UX.showSummaryAnimationDuration) { [self] in
+           gradient.alpha = 0.0
+           tabSnapshot.transform = CGAffineTransform(translationX: 0.0,
+                                                     y: view.frame.height -
+                                                     UX.tabSnapshotFinalPositionBottomPadding -
+                                                     tabSnapshotOffset)
+           loadingLabel.alpha = 0.0
+           summaryView.alpha = 1.0
+           closeButton.alpha = 1.0
+           view.backgroundColor = theme.colors.layer1
+        } completion: { [weak self] _ in
+            if let snapshot = self?.tabSnapshot {
+                UIView.animate(withDuration: 0.2) {
+                    self?.view.bringSubviewToFront(snapshot)
+                }
+            }
+        }
+    }
+
+    // MARK: - PanGesture
+
+    @objc
+    private func onTabSnapshotPan(_ gesture: UIPanGestureRecognizer) {
+        let translationY = gesture.translation(in: view).y
+        let tabSnapshotOffset = tabSnapshotTopConstraint?.constant ?? 0.0
+        let imageViewTransform = CGAffineTransform(translationX: 0.0,
+                                                   y: view.frame.height
+                                                   - UX.tabSnapshotFinalPositionBottomPadding -
+                                                   tabSnapshotOffset)
+        switch gesture.state {
```

Comment:
Code under the cases here could be in it's own function, so the switch case is easier to glance at?